packages/shared/src/db/queries.ts

```diff
@@ -1711,17 +1765,33 @@ export const addJoinedGroupChannel = createWriteQuery(
   async ({ channelId }: { channelId: string }, ctx: QueryCtx) => {
     logger.log('addJoinedGroupChannel', channelId);
 
-    await ctx.db.insert($groupNavSectionChannels).values({
-      channelId,
-      groupNavSectionId: 'default',
-    });
-
-    return await ctx.db
+    // First update the channel membership
+    await ctx.db
       .update($channels)
       .set({
         currentUserIsMember: true,
       })
       .where(eq($channels.id, channelId));
+
+    // Then try to add to nav section if not already there
+    const existing = await ctx.db
+      .select()
+      .from($groupNavSectionChannels)
+      .where(
+        and(
+          eq($groupNavSectionChannels.channelId, channelId),
+          eq($groupNavSectionChannels.groupNavSectionId, 'default')
```

Comment:
I'm kinda confused about this part, when would we expect the channel to not already be in a section? How does this still work if it's in a section that isn't `default`?

Reply:
You're right—we should check if the channel exists in any section, then only add it to 'default' if it truly is unassigned. Let me adjust.